### PR TITLE
thanos-ruler: Add thanos ruler alerts

### PR DESCRIPTION
**Pull request by lilic**
Adds thanos alerting rules for thanos ruler component.

### CODE REVIEW

**Comment by lilic** [comment]
/retest

CI failure.

**Comment by lilic** [comment]
/hold cancel

cc @openshift/openshift-team-monitoring PTAL

File: pkg/tasks/thanos_ruler_user_workload.go [hunk 1]
```diff
@@ -254,7 +255,22 @@ func (t *ThanosRulerUserWorkloadTask) create() error {
 	}
 
 	err = t.client.CreateOrUpdateServiceMonitor(trsm)
-	return errors.Wrap(err, "reconciling Thanos Ruler ServiceMonitor failed")
+	if err != nil {
+		return errors.Wrap(err, "reconciling Thanos Ruler ServiceMonitor failed")
+	}
+	fmt.Println("thanos ruler")
```
Comment (simonpasquier) [inline pkg/tasks/thanos_ruler_user_workload.go:261]: to be removed?

File: pkg/tasks/thanos_ruler_user_workload.go [hunk 2]
```diff
@@ -254,7 +255,22 @@ func (t *ThanosRulerUserWorkloadTask) create() error {
...
+	if err != nil {
+		return errors.Wrap(err, "initializing Thanos Ruler PrometheusRule failed")
+	}
+	fmt.Println("thanos ruler prom rule")
+
+	err = t.client.CreateOrUpdatePrometheusRule(pm)
+	if err != nil {
+		return errors.Wrap(err, "reconciling Thanos Ruler PrometheusRule failed")
+	}
+	fmt.Println("thanos ruler prom done")
```
Comment (simonpasquier) [inline pkg/tasks/thanos_ruler_user_workload.go:272]: ditto

File: pkg/tasks/thanos_ruler_user_workload.go (hunk 1 above)
Comment (lilic) [reply pkg/tasks/thanos_ruler_user_workload.go:261]: No words from my side, apart from "don't push before coffee" 🤦‍♂

**Comment by lilic** [comment]
Good catch @simonpasquier can you please have a look again, thanks!

**Comment by s-urbaniak** [comment]
/approve
"Thanos Ruler PrometheusRule" ... i like it :+1:

**Pull request merged**